assets/tests/components/detourListPage.test.tsx

```diff
@@ -1,13 +1,63 @@
-import { describe, test, expect, jest } from "@jest/globals"
+import { describe, test, expect, jest, beforeEach } from "@jest/globals"
 import React from "react"
-import renderer from "react-test-renderer"
 import { DetourListPage } from "../../src/components/detourListPage"
+import { fetchDetours } from "../../src/api"
+import { neverPromise } from "../testHelpers/mockHelpers"
+import { Ok } from "../../src/util/result"
+import { render, screen } from "@testing-library/react"
 
-jest.useFakeTimers().setSystemTime(new Date("2024-08-16T13:00:00"))
+jest.useFakeTimers().setSystemTime(new Date("2024-08-29T20:00:00"))
+
+jest.mock("../../src/api")
+
+beforeEach(() => {
+  jest.mocked(fetchDetours).mockReturnValue(neverPromise())
+})
 
 describe("DetourListPage", () => {
-  test("renders detour list page with dummy data", () => {
-    const tree = renderer.create(<DetourListPage />).toJSON()
-    expect(tree).toMatchSnapshot()
+  test("renders detour list page", async () => {
+    jest.mocked(fetchDetours).mockResolvedValue(
+      Ok({
+        active: [
+          {
+            route: "1",
+            direction: "Inbound",
+            name: "Headsign A",
+            intersection: "Street A & Avenue B",
+            updated_at: 1724866392,
+          },
+          {
+            route: "2",
+            direction: "Outbound",
+            name: "Headsign B",
+            intersection: "Street C & Avenue D",
+            updated_at: 1724856392,
+          },
+        ],
+        draft: null,
```

Comment:
Question: Will `draft` ever be `null`? Or is it more that it could be empty?

Reply:
Wait no, looking at the backend logic, I get why this would be `null` rather than `[]`.